src/main/scala/dpla/ingestion3/executors/EnrichExecutor.scala

```diff
@@ -92,33 +102,65 @@ trait EnrichExecutor extends Serializable {
       .format("com.databricks.spark.avro")
       .save(dataOut)
 
+    // build the information for reporting
+    val endTime = System.currentTimeMillis()
+    val attemptedCount = mappedRows.count()
+
+    val timeSummary = TimeSummary(
+      Utils.formatDateTime(startTime),
+      Utils.formatDateTime(endTime),
+      Utils.formatRuntime(endTime-startTime)
+    )
+
+    val operationSummary = OperationSummary(
+      attemptedCount,
+      improvedCount.count,
+      attemptedCount-improvedCount.count
+    )
+
+    val enrichOpSummary = EnrichmentOpsSummary(
+      typeImprovedCount.count,
+      dateImprovedCount.count,
+      lamgImprovedCount.count,
+      placeImprovedCount.count,
+      langSummary = PrepareEnrichmentReport.generateLanguageReport(messages)
+    )
+
+    val summaryData = EnrichmentSummaryData(shortName, operationSummary, timeSummary, enrichOpSummary)
+
+    // Write warn and error messages to CSV files
+    logger.info(EnrichmentSummary.getSummary(summaryData))
+
     sc.stop()
 
     // Clean up checkpoint directory, created above
     Utils.deleteRecursively(new File("/tmp/checkpoint"))
 
     // Log error messages.
-    failures.foreach(msg => logger.warn(s"Enrichment error >> $msg"))
+    failures.foreach(logger.error(_))
```

Comment:
What does line 140 do?  Does it write to a log file?

Reply:
Exactly